### PR TITLE
Migrate nightly builds and releases to ghcr.io

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

As part of the work to reduce Tekton's infra spend, we are migrating nightly builds and new releases from gcr.io to ghcr.io, to reduce the expensive egress bandwith utilisation
(https://github.com/tektoncd/plumbing/issues/2157).

This PR introduces support for publishing container images to ghcr.io (YAML files are still published to Google Cloud storage). It also replicates a few improvements that have been done in the pipeline release pipeline before, such us:

- Support for hashing the image path (for nicer path in GitHub)
- Support for skipping build and test tasks, to be enabled for nightly builds to save CPU time

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

/kind misc

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/kind misc

**Comment by khrm** [comment]
@savitaashture Let's merge this.

File: tekton/release-pipeline.yaml
Comment (savitaashture) [inline tekton/release-pipeline.yaml:141]: naive query 

I see its still using gcr.io only how it ensure usage of `ghcr.io` :thinking:

Comment (afrittoli) [reply tekton/release-pipeline.yaml:141]: Yes, catalog tasks are still published as bundles to `gcr.io`. 

The bandwidth consumption associated to `Task` bundles is probably very small compared to that of container images, which is why we started with moving container images. However you are right, it would be good to move `Task` bundles as well eventually.

The [publish pipeline](https://github.com/tektoncd/plumbing/blob/main/tekton/resources/cd/catalog-template.yaml) would probably have to be adapted.

**Comment by afrittoli** [comment]
Once this is approved, I will mark packages as public. They are private by default, and should be visible to maintainers https://github.com/orgs/tektoncd/packages?repo_name=triggers

**Review by dibyom — COMMENTED** [review]
/lgtm

**Review by dibyom — APPROVED** [review]

File: tekton/release-pipeline.yaml
Comment (savitaashture) [reply tekton/release-pipeline.yaml:141]: Sure thank you @afrittoli for the clarification

**Pull request merged**
